### PR TITLE
CIV-5681 CCJ check answers

### PR DESCRIPTION
**Pull request by KyleMulvihillV1**
JIRA link

https://tools.hmcts.net/jira/browse/CIV-5681

Change description

Check answers screen for CCJ.

Does this PR introduce a breaking change? (check one with "x")

[ ] Yes
[X] No

### CODE REVIEW

File: src/main/common/form/models/claimantResponse/ccj/ccjPaymentOption.ts [hunk 1]
```diff
@@ -16,4 +16,14 @@ export class CcjPaymentOption {
   isCcjPaymentOptionInstalments() {
     return this.type === PaymentOptionType.INSTALMENTS;
   }
+
+  formatOptionType(type: PaymentOptionType) {
+    if(type === PaymentOptionType.INSTALMENTS) {
+      return 'By instalments';
```
Comment (TatsyMaslova) [inline src/main/common/form/models/claimantResponse/ccj/ccjPaymentOption.ts:22]: what about translation of these to welsh?

File: src/main/modules/i18n/locales/en.json [hunk 2]
```diff
@@ -422,6 +434,7 @@
       "INCOME_WORKING_TAX_CREDIT": "Working Tax Credit",
       "OWE_MONEY": "Do you owe the money claimed?",
       "PAYMENT_FREQUENCY": "Frequency of payments",
+      "PAYMENT_TITLE": "Payment",
```
Comment (portizdev) [inline src/main/modules/i18n/locales/en.json:437]: Make sure you always use the same lines in the en and cy files.

**Review by portizdev — CHANGES_REQUESTED** [review]

File: src/main/services/features/claim/checkAnswers/financialSection/buildPaymentSection.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    if (ccjPaymentOption.model.isCcjPaymentOptionInstalments() && claim.claimantResponse?.ccjRequest?.repaymentPlanInstalments) {
+      paymentDetailsSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.CCJ_INSTALMENTS_PAYMENT_FREQUENCY', {lng}),
+        formatPaymentFrequency(claim.claimantResponse.ccjRequest.repaymentPlanInstalments.paymentFrequency)));
+    }
+  }
+
+  function formatPaymentFrequency(paymentFrequency: TransactionSchedule) {
+    if(paymentFrequency === TransactionSchedule.WEEK){
+      return 'Each week';
```
Comment (portizdev) [inline src/main/services/features/claim/checkAnswers/financialSection/buildPaymentSection.ts:71]: Same as Tanya says. Use always key translations.

File: src/main/services/features/claim/checkAnswers/financialSection/buildPaymentSection.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  if(claim.claimantResponse?.ccjRequest?.ccjPaymentOption?.type) {
+    const ccjPaymentOption = new GenericForm(new CcjPaymentOption(claim.claimantResponse.ccjRequest.ccjPaymentOption.type));
+    paymentDetailsSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.CCJ_HOW_TO_PAY', {lng}),
+      ccjPaymentOption.model.formatOptionType(claim.claimantResponse.ccjRequest.ccjPaymentOption.type)));
+
+    if (ccjPaymentOption.model.isCcjPaymentOptionBySetDate() && claim.claimantResponse?.ccjRequest?.defendantPaymentDate?.date) {
+      paymentDetailsSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.CCJ_WHEN_DO_YOU_WANT_TO_BE_PAID_BY', {lng}),
+        formatDateToFullDate(claim.claimantResponse.ccjRequest.defendantPaymentDate.date)));
```
Comment (laperla11) [inline src/main/services/features/claim/checkAnswers/financialSection/buildPaymentSection.ts:50]: you need to pass the `lng` as a second argument to translate month

File: src/main/services/features/claim/checkAnswers/financialSection/buildPaymentSection.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    if (ccjPaymentOption.model.isCcjPaymentOptionInstalments() && claim.claimantResponse?.ccjRequest?.repaymentPlanInstalments) {
+      paymentDetailsSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.CCJ_INSTALMENTS_OF', {lng}),
+        currencyFormatWithNoTrailingZeros(claim.claimantResponse.ccjRequest.repaymentPlanInstalments.amount)));
+    }
+
+    if (ccjPaymentOption.model.isCcjPaymentOptionInstalments() && claim.claimantResponse?.ccjRequest?.repaymentPlanInstalments?.firstPaymentDate) {
+      paymentDetailsSection.summaryList.rows.push(summaryRow(t('PAGES.CHECK_YOUR_ANSWER.CCJ_FIRST_PAYMENT_DATE', {lng}),
+        formatDateToFullDate(claim.claimantResponse.ccjRequest.repaymentPlanInstalments.firstPaymentDate.date)));
```
Comment (laperla11) [inline src/main/services/features/claim/checkAnswers/financialSection/buildPaymentSection.ts:60]: you need to pass the `lng` as a second argument to translate month

**Review by laperla11 — CHANGES_REQUESTED** [review]

**Review by laperla11 — APPROVED** [review]

File: src/test/unit/services/features/claim/checkAnswers/financialSection/buildPaymentSection.test.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+const INSTALMENTS_OF = '£200';
+
+describe('Payment Section', () => {
+
+  it('should return your Payment summary sections for pay by set date', async () => {
+    //When
+    const claim = createCCJClaimWithClaimResponseDetailsForPayBySetDate();
+    const summarySections = await getSummarySections(constVal.CLAIM_ID, claim, 'cimode');
+    //Then
+    expect(summarySections.sections[constVal.INDEX_CCJ_PAYMENT_SECTION].title).toBe('PAGES.CHECK_YOUR_ANSWER.PAYMENT_TITLE');
```
Comment (TatsyMaslova) [inline src/test/unit/services/features/claim/checkAnswers/financialSection/buildPaymentSection.test.ts:32]: didn't Sampan say it slows down the tests when we are importing everything form constVal? i remember when he was fixing unit tests he was removing it

File: src/test/utils/checkAnswersConstants.ts [hunk 7]
```diff
@@ -11,6 +10,8 @@ export const INDEX_RESPONSE_CLAIM_SECTION_PART_ADMISSION = 3;
 export const INDEX_FINANCIAL_SECTION_PART_ADMISSION = 5;
 export const INDEX_RESPONSE_PAYMENT_SECTION = 6;
 export const INDEX_RESPONSE_FREE_TELEPHONE_MEDIATION_SECTION = 7;
+export const INDEX_CCJ_THEIR_DETAILS_SECTION = 0;
+export const INDEX_CCJ_PAYMENT_SECTION = 1;
```
Comment (TatsyMaslova) [inline src/test/utils/checkAnswersConstants.ts:14]: i think easier not to add this but just use the index 0 and 1 in the test to make it quicker

**Review by TatsyMaslova — APPROVED** [review]

**Review by portizdev — APPROVED** [review]

**Pull request merged**